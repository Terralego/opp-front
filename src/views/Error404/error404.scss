@import '../../variables.scss';

.error-404 {
  display: flex;
  flex-direction: column;
  justify-content: center;
  background: $solidgray;
}

.error-404--message,
.error-404--title {
  text-align: center;
  color: #fff;
}

.error-404--message {
  font-size: 2em;
}
