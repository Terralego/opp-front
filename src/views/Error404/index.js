import React from 'react';
import { NavLink } from 'react-router-dom';
import { H1 } from '@blueprintjs/core';
import { Trans, useTranslation } from 'react-i18next';

import './error404.scss';

const Error404View = () => {
  const { t } = useTranslation();
  return (
    <div className="error-404">
      <H1 className="error-404--title">{t('error.404.title')}</H1>
      <p className="error-404--message">
        <Trans i18nKey="error.404.message">
          Go back to <NavLink to="/">homepage</NavLink>
        </Trans>
      </p>
    </div>
  );
};

export default Error404View;
