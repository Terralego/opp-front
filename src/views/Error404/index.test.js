import React from 'react';
import renderer from 'react-test-renderer';

import Error404 from '.';

jest.mock('react-router-dom', () => ({
  NavLink: props => <span {...props} />,
}));

jest.mock('@blueprintjs/core', () => ({
  H1: () => <div>blueprint h1</div>,
}));

it('should render', () => {
  const tree = renderer.create((
    <Error404 />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
