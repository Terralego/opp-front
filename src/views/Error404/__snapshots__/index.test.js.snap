// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="error-404"
>
  <div>
    blueprint h1
  </div>
  <p
    className="error-404--message"
  >
    Go back to 
    <span
      to="/"
    >
      homepage
    </span>
  </p>
</div>
`;
